File: app/src/main/java/com/velaphi/untamed/features/animalDetails/models/Predator.java
package com.velaphi.untamed.features.animalDetails.models;

import android.os.Parcel;
import android.os.Parcelable;

import com.google.firebase.firestore.IgnoreExtraProperties;

@IgnoreExtraProperties
public class Predator implements Parcelable {
    private String name;
    private String image;

    public Predator() {
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public String getImage() {
        return image;
    }

    public void setImage(String image) {
        this.image = image;
    }

    protected Predator(Parcel in) {
        name = in.readString();
        image = in.readString();
    }

    public static final Creator<Predator> CREATOR = new Creator<Predator>() {
        @Override
        public Predator createFromParcel(Parcel in) {
            return new Predator(in);
        }

        @Override
        public Predator[] newArray(int size) {
            return new Predator[size];
        }
    };

    @Override
    public int describeContents() {
        return 0;
    }

    @Override
    public void writeToParcel(Parcel dest, int flags) {
        dest.writeString(name);
        dest.writeString(image);
    }
}

File: app/src/main/java/com/velaphi/untamed/features/animalDetails/models/Predator.kt
package com.velaphi.untamed.features.animalDetails.models

import android.os.Parcel
import android.os.Parcelable
import com.google.firebase.firestore.IgnoreExtraProperties

@IgnoreExtraProperties
class Predator : Parcelable {
    var name: String? = null
    var image: String? = null

    constructor() {}

    protected constructor(`in`: Parcel) {
        name = `in`.readString()
        image = `in`.readString()
    }

    override fun describeContents(): Int {
        return 0
    }

    override fun writeToParcel(dest: Parcel, flags: Int) {
        dest.writeString(name)
        dest.writeString(image)
    }

    companion object CREATOR : Parcelable.Creator<Predator> {
        override fun createFromParcel(parcel: Parcel): Predator {
            return Predator(parcel)
        }

        override fun newArray(size: Int): Array<Predator?> {
            return arrayOfNulls(size)
        }
    }

}